src/v/iceberg/rest_client/oauth_token.h

```diff
@@ -9,7 +9,10 @@ namespace iceberg::rest_client {
 // Oauth token returned by the catalog server, in exchange for credentials
 struct oauth_token {
     ss::sstring token;
+    ss::sstring token_type;
```

Comment:
This feels like there is some enum here - can we add a comment about what kind of values are expected here?